### PR TITLE
add unit test for preprocessor and javagen

### PR DESCRIPTION
**Pull request by XiaofeiCao**
This is related to https://github.com/Azure/autorest.java/issues/1256

### CODE REVIEW

File: preprocessor/src/test/java/com/azure/autorest/PreprocessorTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ */
+public class PreprocessorTest {
+
+    @Test
+    @Ignore("no validation")
+    public void processTest() throws Exception {
+        MockPreprocessor preprocessor = new MockPreprocessor(new Connection(System.out, System.in), "dummy", "dummy");
+        // This file can be obtained by calling generate commands from `generate` or `generate.bat`, it will appear under this project
+        // It's a debug file from the output of modelerfour during the generate process
+        String file = readFile("code-model.yaml");
```
[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply preprocessor/src/test/java/com/azure/autorest/PreprocessorTest.java]: Sounds fair, thanks for pointing that out!

File: preprocessor/src/test/java/com/azure/autorest/PreprocessorTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                                                          Tag customTag) {
+                // if value of property is null, ignore it.
+                if (propertyValue == null) {
+                    return null;
+                }
+                else {
+                    return super.representJavaBeanProperty(javaBean, property, propertyValue, customTag);
+                }
+            }
+        };
```
[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply preprocessor/src/test/java/com/azure/autorest/PreprocessorTest.java]: Refactored the code to move that code from the main flow of the test into the MockPreprocessor. Will definitely extract a utils method if it's used more than once.

File: preprocessor/src/test/java/com/azure/autorest/PreprocessorTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply preprocessor/src/test/java/com/azure/autorest/PreprocessorTest.java]: added

**Comment by weidongxu-microsoft** [comment]
try not to make a too big PR. keep it manageable, then add new iterations as new PRs.

**Comment by XiaofeiCao** [comment]
Linked two separate PRs to resolve the issue. Closing this pr. 
https://github.com/Azure/autorest.java/pull/1286
https://github.com/Azure/autorest.java/pull/1286

**Pull request closed without merging**